utils/Stats.hs

```diff
@@ -15,16 +16,21 @@ data Histogram = H {
     , collision     :: !Int
     } deriving Show
 
-instance Monoid Histogram where
-    mempty = H 0 0 0 0 0
-    mappend h1 h2 = H {
+instance Semigroup Histogram where
+    h1 <> h2 = H {
           empty         = empty h1 + empty h2
         , leaf          = leaf h1 + leaf h2
         , bitmapIndexed = bitmapIndexed h1 + bitmapIndexed h2
         , full          = full h1 + full h2
         , collision     = collision h1 + collision h2
         }
 
+instance Monoid Histogram where
+    mempty = H 0 0 0 0 0
+#if __GLASGOW_HASKELL__ < 803
```

Comment:
shouldn't this been `MIN_VERSION_base` check

Reply:
Indeed. I prioritized maintaining the existing code conventions over my desire for a slightly more correct macro :)

Of course, that's not to say we shouldn't switch over to `MIN_VERSION_base` at some point.

Reply:
`MIN_VERSION_base` would surely be less error-prone (and statically checked for form!), and the original reason for using `__GLASGOW_HASKELL__` is becoming more and more moot.

Reply:
Since we are the subject, I understand the reasons for using `MIN_VERSION_base` instead of  `__GLASGOW_HASKELL__` on the above example, but what should I use when using CPP on the `Prelude`?

For example, for hiding `Prelude.<>` I can use

```
#if MIN_VERSION_base(4,11,0)
import Prelude hiding ((<>))
#endif
```

or

```
#if __GLASGOW_HASKELL__ >= 804
import Prelude hiding ((<>))
#endif
```
 
I think I should use `__GLASGOW_HASKELL__ ` because the `Prelude` is defined by GHC, but I would to know your opinions about it.

Reply:
`Prelude` is defined by `base`. AMP, FTP and now SMP are all library changes, and should be guarded with `MIN_VERSION_base`.

From top of my head, you want to use `__GLASGOW_HASKELL__` around specific `OPTIONS_GHC` or language pragmas, like `GeneralizedNewtypeDeriving`.

Sometimes you *should* guard on both, e.g. with `DeriveGeneric`, as you need both compiler (GHC) and library (base) support. But in practice guarding on GHC is enough, as "if GHC supports `DeriveGeneric`, there is `GHC.Generics` in `base`".

Reply:
Crystal. Thanks!